sycl/include/CL/sycl/detail/aligned_allocator.hpp

```diff
@@ -0,0 +1,71 @@
+//==------------ aligned_allocator.hpp - SYCL standard header file ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/cl.h>
+#include <CL/sycl/detail/cnri.h>
+#include <CL/sycl/range.hpp>
+
+#include <cstring>
+#include <memory>
+#include <vector>
+
+namespace cl {
+namespace sycl {
+template <typename T, size_t Alignment>
+class aligned_allocator {
+public:
+  using value_type = T;
+  using pointer = T*;
+  using const_pointer = const T*;
+  using reference = T&;
+  using const_reference = const T&;
+
+public:
+  template<typename U>
+  struct rebind {
+    typedef aligned_allocator<U, Alignment> other;
+  };
+
+  aligned_allocator() = default;
+  aligned_allocator(const aligned_allocator&) = default;
+
+  void construct(pointer Ptr, const_reference Val) {
+    new (Ptr) value_type(Val);
+  }
+
+  void destroy(pointer Ptr) { Ptr->~value_type(); }
+
+  pointer address(reference Val) const { return &Val; }
+  const_pointer address(const_reference Val) { return &Val; }
+
+  pointer allocate(size_t Size) {
+    Size += Alignment - Size % Alignment;
+    pointer Result = reinterpret_cast<pointer>(
+      aligned_alloc(Alignment, Size * sizeof(value_type)));
```

Comment:
Is this `std::aligned_alloc` from C++17? If so, I am pleased by such modernism, but please add `std::`.

Reply:
more likely to be the C version from C11: https://en.cppreference.com/w/c/memory/aligned_alloc which I don't think rests in the std namespace (at least from the looks of it). I think the runtime is strictly C++11 just now? I could be wrong though

Reply:
Yes, this is why I was surprised.
But then we should have a `#include <cstdlib>`?
By looking at the GCC 8 `libstdc++`, they use `::aligned_aloc` to refer to the C version, which included with `#include <stdlib.h>`

Reply:
Unfortunately we are doing C++11 now :(. As far as I google, ::aligned_alloc is supported from GCC7 (still C++17 feature), and yet it's not good practice to increase expectations from user's compiler. LLVM itself just moved from GCC4 to GCC5.

Reply:
So yes, it's C11 aligned_alloc.

Reply:
OK, let's stay in the Jurassic world for now... :-)